File: Python/Leetcode/Array/jumpGameII.py
class Solution:
    def jumpGameII(self, nums: list[int]):
        l = r = 0
        res = 0
        
        while r < len(nums)-1:
            furthest = 0
            for i in range(l, r + 1):
                furthest = max(furthest, i + nums[i])
            
            l = r + 1
            r = furthest
            res += 1
        
        return res
                
if __name__=='__main__':
    solution_instance = Solution()
    solution_instance.jumpGameII([2,3,0,1,4])

File: C++/Leetcode/Array/jumpGameII.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
    public:
        int jumpGameII(std::vector<int>& nums){
            int l = 0, r = 0;
            int res = 0;

            while (r < nums.size()-1){
                int furthest = 0;
                
                for (int i = l; i <= r; ++i) {
                    furthest = std::max(furthest, nums[i] + i);
                }
                l = r + 1;
                r = furthest;
                res++;
            }
            return res;
        }
};

int main(){
    Solution solution_instance;
    std::vector<int> nums = {2,3,1,1,4};
    std::cout << solution_instance.jumpGameII(nums) << std::endl;
    return 0;
}